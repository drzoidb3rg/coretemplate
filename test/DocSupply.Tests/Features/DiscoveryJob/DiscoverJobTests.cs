using DocSupply.Features.DiscoveryJob;
using NUnit.Framework;
using Shouldly;

namespace DocSupply.Tests.Features.DiscoveryJob
{
    [TestFixture]
    public class DiscoverJobTests
    {
        public DiscoverJobTests()
        {
            WebTest.WithDiffTool();

        }

        [TestCase]
        public void DiscoverJobTests_PostRisFilesJSON()
        {
            using (var store = WebTest.GetDocumentStore())
            {
                using (var _ = Tester.AsUserTwo()
                    .WithStore(store)
                    .WithIndex(new DiscoveryJobRecordsByUser())
                    .PostFile("/DiscoveryJobList", @"Features\DiscoveryJob\example_export_from_EPPI.ris", "file one")
                    .AsJson()
                    .Get("/DiscoveryJobList"))
                {
                    _.PrettyJsonResponse().ShouldMatchApproved(_.WithApprovalFolder);
                }
            }
        }

        [TestCase]
        public void DiscoverJobTests_PostRisFiles()
        {
            using (var store = WebTest.GetDocumentStore())
            {
                using (Tester.AsUserTwo()
                    .WithStore(store)
                    .PostFile("/DiscoveryJobList", @"Features\DiscoveryJob\example_export_from_EPPI.ris", "file one"))

                using (var _ = Tester.AsUserOne()
                   .WithStore(store)
                   .WithIndex(new DiscoveryJobRecordsByUser())
                   .PostFile("/DiscoveryJobList", @"Features\DiscoveryJob\example_export_from_EPPI.ris", "file one")
                   .PostFile("/DiscoveryJobList", @"Features\DiscoveryJob\example_export_from_EPPI.ris", "file two")
                   .Get("/DiscoveryJobList"))
                {
                    _.ResponseString().ShouldMatchApproved(_.WithApprovalFolder);
                };
            }
        }


        [TestCase]
        public void DiscoverJobTests_PostInvalidRisFile()
        {
            using (var store = WebTest.GetDocumentStore())
            {
                using (var _ = Tester.AsUserOne()
                   .WithStore(store)
                   .PostFile("/DiscoveryJobList", @"Features\DiscoveryJob\example_invalid_file_format.ris", "some file")
                   .IsRedirect())
                {
                    var doc = _.GetFromStore<DiscoveryJobRecord>("DiscoveryJobRecord/1");
                    Assert.AreEqual(null, doc);
                };
            }
        }

        [TestCase]
        public void DiscoverJobTests_PostRisFileWithParagraphs()
        {
            using (var store = WebTest.GetDocumentStore())
            {
                using (var _ = Tester.AsUserOne()
                   .WithStore(store)
                   .PostFile("/DiscoveryJoblist", @"Features\DiscoveryJob\ris_file_with_paragraphs.txt", "some file")
                   .IsRedirect())
                {
                    var doc = _.GetFromStore<DiscoveryJobRecord>("DiscoveryJobRecord/1");
                    Assert.AreEqual("some file", doc.FileName);
                    Assert.AreEqual(true, doc.Risfile.Valid);
                };
            }
        }

    }
}
