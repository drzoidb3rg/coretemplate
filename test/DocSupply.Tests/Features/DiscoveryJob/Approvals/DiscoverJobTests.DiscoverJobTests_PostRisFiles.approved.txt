
<p><h2>Discovery job list</h2></p>


    <p>Upload ris file to create a new discovery job</p>
    <form method="post" enctype="multipart/form-data" action="/discoveryjoblist">
        <input type="file" name="files" multiple />
        <input type="submit" value="Upload" disabled />
    </form>


<h3>List of jobs</h3>

    <div><a href="discoveryjob/2">file one</a> user one</div> 
    <div><a href="discoveryjob/3">file two</a> user one</div> 



